src/utils/daos_hdlr.c

```diff
@@ -2566,95 +2576,241 @@ fs_copy(struct file_dfs *src_file_dfs,
+	dm->src_coh = DAOS_HDL_INVAL;
+	dm->dst_poh = DAOS_HDL_INVAL;
+	dm->dst_coh = DAOS_HDL_INVAL;
+	dm->cont_prop_oid = DAOS_PROP_CO_ALLOCED_OID;
+	dm->cont_prop_layout = DAOS_PROP_CO_LAYOUT_TYPE;
+	dm->cont_layout = DAOS_PROP_CO_LAYOUT_UNKOWN;
+	dm->cont_oid = 0;
+
+}
+
 static int
-fs_copy_connect(struct file_dfs *src_file_dfs,
-		struct file_dfs *dst_file_dfs,
-		struct fs_copy_args *fa,
-		char *sysname,
-		daos_cont_info_t *src_cont_info,
-		daos_cont_info_t *dst_cont_info,
-		struct duns_attr_t *src_dattr,
-		struct duns_attr_t *dst_dattr)
+dm_get_cont_prop(daos_handle_t coh,
+		 char *sysname,
+		 daos_cont_info_t *cont_info,
+		 uint32_t dpe_type,
+		 uint64_t *dpe_val)
+{
+	int                     rc = 0;
+	daos_prop_t		*prop = daos_prop_alloc(1);
+
+	if (prop == NULL) {
+		fprintf(stderr, "Failed to allocate prop (%d)", rc);
+		D_GOTO(out, rc);
+	}
+
+	prop->dpp_entries[0].dpe_type = dpe_type;
+
+	rc = daos_cont_query(coh, NULL, prop, NULL);
+	if (rc) {
+		fprintf(stderr, "daos_cont_query() failed (%d)", rc);
+		daos_prop_free(prop);
+		D_GOTO(out, rc);
+	}
+
+	*dpe_val = prop->dpp_entries[0].dpe_val;
+	daos_prop_free(prop);
+out:
+	return rc;
+}
+
+static int
+dm_connect(bool is_posix_copy,
+	   struct file_dfs *src_file_dfs,
+	   struct file_dfs *dst_file_dfs,
+	   struct dm_args *ca,
+	   char *sysname,
+	   char *path,
+	   daos_cont_info_t *src_cont_info,
+	   daos_cont_info_t *dst_cont_info)
 {
 	/* check source pool/conts */
-	int rc = 0;
+	int			rc = 0;
+	struct duns_attr_t	dattr = {0};
+	daos_prop_t		*props = NULL;
+	dfs_attr_t		attr;
 
 	/* open src pool, src cont, and mount dfs */
 	if (src_file_dfs->type == DAOS) {
-		rc = daos_pool_connect(fa->src_p_uuid, sysname,
-				       DAOS_PC_RW, &fa->src_poh, NULL, NULL);
+		rc = daos_pool_connect(ca->src_p_uuid, sysname,
+				       DAOS_PC_RW, &ca->src_poh, NULL, NULL);
 		if (rc != 0) {
 			fprintf(stderr, "failed to connect to destination "
 				"pool: %d\n", rc);
 			D_GOTO(out, rc);
 		}
-		rc = daos_cont_open(fa->src_poh, fa->src_c_uuid, DAOS_COO_RW,
-				    &fa->src_coh, src_cont_info, NULL);
+		rc = daos_cont_open(ca->src_poh, ca->src_c_uuid, DAOS_COO_RW,
+				    &ca->src_coh, src_cont_info, NULL);
 		if (rc != 0) {
 			fprintf(stderr, "failed to open source "
 				"container: %d\n", rc);
 			D_GOTO(err_src_root, rc);
 		}
-		rc = dfs_mount(fa->src_poh, fa->src_coh, O_RDWR,
-			       &src_file_dfs->dfs);
-		if (rc) {
-			fprintf(stderr, "dfs mount on source failed: %d\n", rc);
-			D_GOTO(err_src_dfs, rc);
+		if (is_posix_copy) {
+			rc = dfs_mount(ca->src_poh, ca->src_coh, O_RDWR,
+				       &src_file_dfs->dfs);
+			if (rc) {
+				fprintf(stderr, "dfs mount on source "
+					"failed: %d\n", rc);
+				D_GOTO(err_src, rc);
+			}
 		}
 	}
 
+	/* get source container type */
+	rc = dm_get_cont_prop(ca->src_coh, sysname, src_cont_info,
+			      ca->cont_prop_layout, &ca->cont_layout);
+	if (rc != 0) {
+		fprintf(stderr, "failed to query source "
```

Comment:
is there a reason you don't fail here and abort vs just print error and continue?

Reply:
@mchaarawi well originally it was because the filesystem copy uses this connect call, and it would be ok for the query to fail if the source was not a DAOS container, but a POSIX file. I changed this to only query the source container if it is not of type POSIX.

Reply:
if it is OK to fail then we should not print errors to stderr.
but i don't get why get prop would fail and why it would be ok to fail, whatever the container type is

Reply:
@mchaarawi well I mean in this case if src_mfu_file->type == POSIX it is not a DAOS container, the source would be coming from the POSIX filesystem. I updated this to check if that was the source, so it skips the container query in that case.